### PR TITLE
Added language injection into `#!/shebang` recipes

### PR DESCRIPTION
**Pull request by mkatychev**
Shebang files now pick up the language to use for syntax highlighting of inner types.

### CODE REVIEW

**Comment by mkatychev** [comment]
One issue with the approach in this PR is that interpolation nodes are overwritten so that it becomes simple a string:

https://github.com/IndianBoy42/tree-sitter-just/blob/53ab642becaf77553412d6f04bc8aaa5352cd890/test/corpus/recipes.txt#L366-L369

```scm
      (command ; [4:5 - 22]
        name: (command_name ; [4:5 - 8]
          (word)) ; [4:5 - 8]
        argument: (string ; [4:10 - 22]
          (string_content)))) ; [4:11 - 21]
```

Some advice could be used to layer the various languages here.
I've tried to use `injection.parent` here but that does not seem to make the `interpolation` node show up in the tree:
https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection

**Comment by tgross35** [comment]
Thanks for the PR! I'll take a better look at this tomorrow. Shebangs are pretty tricky, I had something that sort of worked on the `next` branch here https://github.com/IndianBoy42/tree-sitter-just/blob/4c5c1f21ff871d57141d83dc606e433895698c42/queries-src/injections.scm, but that's quite a bit out of date by now.

**Comment by mkatychev** [comment]
The interpolation issue is quite similar to here:
https://github.com/nvim-treesitter/nvim-treesitter/issues/3110

Would be nice if we could specify something similar to `prec(number, rule)` for injections so that not all of the CST gets overridden.

File: grammar.js [hunk 1]
```diff
@@ -274,8 +274,7 @@ module.exports = grammar({
       seq(
         $._indent,
         optional(seq($.shebang, $._newline)),
-        repeat(choice(seq($.recipe_line, $._newline), $._newline)),
-        $._newline,
+        repeat1(seq($.recipe_line, $._newline)),
```
Comment (mkatychev) [inline grammar.js:277]: This was needed so that non-newline separated recipes would have comments identified correctly:

```justfile
# foo
foo:
  echo "foo"
# bar
bar:
  echo "bar"
```

Comment (tgross35) [reply grammar.js:277]: Awesome, thank you. Would you mind splitting this change to a separate PR with a new test? Just so I can merge it right away (if not, I can split it off)

File: queries/just/textobjects.scm [hunk 2]
```diff
@@ -9,6 +9,9 @@
 (dependency_expression
   (_) @parameter.inside) @parameter.outer
 
+(function_call
+  arguments: (sequence (expression) @parameter.inside) @parameter.outer) @function.outer
+
```
Comment (tgross35) [inline queries/just/textobjects.scm:14]: Was this change intended? Any sort of query changes should go in `queries-src`, this file gets autogenerated

File: queries-src/injections.scm [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
+  (#set! injection.include-children)
+) @injection.content
+
+(comment) @comment
+
+
+(source_file
+    ; (item (setting lang:(identifier) @language))
+    (recipe (recipe_body) @content)
+)
```
Comment (tgross35) [inline queries-src/injections.scm:15]: FYI the scheme style has the closing parenthesis on the same line as whatever it closes. I know it looks pretty weird, but at least it's consistent.

Also a loose comment. No big deal, I'll fix this up before applying.

File: grammar.js [hunk 4]
```diff
@@ -287,7 +286,17 @@ module.exports = grammar({
...
+
+    shebang: ($) =>
+      seq(
+        "#!",
+        /.*\//,
+        optional(seq("env ", repeat($._shebang_flag))),
+        // defer dilemma: python3 => python, but json5 => json5
+        alias($.identifier, $.language),
+        repeat($._shebang_flag),
+      ),
```
[the remark this answers is not in the document]
Comment (tgross35) [reply grammar.js:299]: Oh, I see you have the ~/local` test. An additional test with just `#!/bin/bash` would still be good as a sanity check, but I can add that before committing.

**Comment by tgross35** [comment]
Some small comments, but overall I think this looks good. I need to actually test it. Please rebase & squash when you add the changes.

Not sure why windows CI is failing, I'll try to fix that before merging this.

File: queries-src/injections.scm (hunk 3 above)
Comment (mkatychev) [reply queries-src/injections.scm:15]: This was an unintended addition when resolving merge conflicts, I'm going to remove since the point of this PR was to defer injections for things _without_ an explicit _and_ inside a recipe 😅 .

File: grammar.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (mkatychev) [reply grammar.js:299]: would you clarify the additional test desired?

File: queries/just/textobjects.scm (hunk 2 above)
Comment (mkatychev) [reply queries/just/textobjects.scm:14]: Nope, I must have merged poorly.

File: grammar.js (hunk 1 above)
Comment (mkatychev) [reply grammar.js:277]: https://github.com/IndianBoy42/tree-sitter-just/pull/119

File: queries/just/textobjects.scm (hunk 2 above)
Comment (mkatychev) [reply queries/just/textobjects.scm]: Update, this call seems to be produced automatically by running `just gen` on the current `HEAD` of this PR

File: test/corpus/recipes.txt [hunk 5]
```diff
@@ -318,6 +318,46 @@ foo:
       (recipe_line
         (text)))))
 
+================================================================================
+shebang: env -i bash --
+================================================================================
+
+foo:
+  #!/usr/local/bin/env -i bash
```
Comment (mkatychev) [inline test/corpus/recipes.txt:326]:
```suggestion
  #!/usr/local/bin/env -i bash --
```

**Comment by tgross35** [comment]
Don't worry I'm looking at this, just trying to figure out the Windows thing :)

File: grammar.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (tgross35) [reply grammar.js:300]: You're right it does, that works for me

**Review by tgross35 — APPROVED** [review]
Guess this doesn't have the CI break anymore, thank you!

**Comment by mkatychev** [comment]
> Guess this doesn't have the CI break anymore, thank you!

@tgross35  good thing you suggested to break up this PR!
